core/src/main/java/com/datastrato/gravitino/listener/impl/EventListenerManager.java

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener.impl;
+
+import com.datastrato.gravitino.listener.EventListenerPlugin;
+import com.datastrato.gravitino.listener.SupportsAsync;
+import com.datastrato.gravitino.listener.SupportsAsync.Mode;
+import com.datastrato.gravitino.utils.MapUtils;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * EventListenerManager loads listeners according to the configurations, and assemble the listeners
+ * with following rules:
+ *
+ * <p>Wrap all listener with EventListenerWrapper to do some common process, like exception handing,
+ * record metrics.
+ *
+ * <p>For async listeners with the shared dispatcher, will create a default AsyncQueueListener to
+ * assemble the corresponding EventListenerWrappers.
+ *
+ * <p>For async listeners with the isolated dispatcher, will create a separate AsyncQueueListener
+ * for each EventListenerWrapper.
+ */
+public class EventListenerManager {
+  private static final Logger LOG = LoggerFactory.getLogger(EventListenerManager.class);
+  public static final String GRAVITINO_EVENT_LISTENER_PREFIX = "gravitino.eventListener.";
+  @VisibleForTesting static final String GRAVITINO_EVENT_LISTENER_NAMES = "names";
+  @VisibleForTesting static final String GRAVITINO_EVENT_LISTENER_CLASSNAME = "className";
+  private static final String GRAVITINO_EVENT_LISTENER_QUEUE_CAPACITY = "queueCapacity";
+  private static final Splitter splitter = Splitter.on(",");
+  private static final Joiner DOT = Joiner.on(".");
+
+  private int queueCapacity;
+  private List<EventListenerPlugin> eventListeners;
+
+  public void init(Map<String, String> properties) {
+    String queueCapacity = properties.get(GRAVITINO_EVENT_LISTENER_QUEUE_CAPACITY);
+    this.queueCapacity =
+        Optional.ofNullable(queueCapacity)
+            .map(capacity -> Integer.valueOf(capacity))
+            .orElse(Integer.valueOf(1000))
```

Comment:
Can you use a class variable to define this, do not hard code here.

Reply:
done